gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/entity/EntityServiceEntityFetcher.java

```diff
@@ -56,6 +59,18 @@ private EntityQueryRequest buildRequest(
     addGroupBys(exploreRequest, builder);
     addSelections(requestContext, exploreRequest, builder);
 
+    // TODO: Push group by down to EQS
+    // If there is a group by, specify a large limit and track actual limit, offset and order by
+    // expression list, so we can compute these once the we get the results.
+    if (requestContext.hasGroupBy()) {
+      // Will need to do the ordering, limit and offset ourselves after we get the group by results
+      builder.setLimit(QueryServiceClient.DEFAULT_QUERY_SERVICE_GROUP_BY_LIMIT);
+      requestContext.setOrderByExpressions(getRequestOrderByExpressions(exploreRequest));
+    } else {
+      // No Group By
+      throw new RuntimeException("Entity request handler only supports group by requests");
```

Comment:
Why? What would fail in this path (other than us not assigning the requested limit and order by)?